src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -333,18 +332,17 @@ static std::vector<Widget> _window_editor_object_selection_widgets = {
                 WidgetInvalidate(*this, WIDX_FILTER_TEXT_BOX);
             }
 
-            for (WidgetIndex i = WIDX_FILTER_RIDE_TAB_TRANSPORT; i <= WIDX_FILTER_RIDE_TAB_STALL; i++)
-            {
-                if (!IsWidgetPressed(i))
-                    continue;
+            if (GetSelectedObjectType() != ObjectType::Ride)
+                return;
 
-                frame_no++;
-                if (frame_no >= window_editor_object_selection_animation_loops[i - WIDX_FILTER_RIDE_TAB_TRANSPORT])
-                    frame_no = 0;
+            if (_selectedSubTab == 0)
+                return;
```

Comment:
The reason for this condition isn't immediately clear when reading the code. Why is tab 0 skipped here? This is the "all" tab, if I get this right?

Reply:
Thanks. Yes, that is indeed the 'all rides' tab. This is completely refactored by #22352, though, so it won't be a problem for much longer.